### PR TITLE
Improve code styling

### PR DESCRIPTION
**Pull request by jlgasparrini**
### In order to improve the code style I install [eslint](https://github.com/typescript-eslint/typescript-eslint) library with typescript support to the project.

Now running `yarn lint` we can take care of any possible code style errors and warnings.

image

Also, I upgrade from typescript `3.1.1` to `4.0.5` to have compatibility with the latest eslint version and to have [these features](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html) available in the project.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    "env": {
+        "browser": true,
+        "es2021": true
+    },
+    "extends": [
+        "standard"
```
Comment (edudepetris) [inline .eslintrc.json]: @jlgasparrini 
following the official doc, two extra lines are include here: notice the extends part

```js
module.exports = {
    "env": {
        "browser": true,
        "es2021": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/recommended"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": 12,
        "sourceType": "module"
    },
    "plugins": [
        "@typescript-eslint"
    ],
    "rules": {
    }
};
```

File: package.json [hunk 2]
```diff
@@ -27,17 +27,25 @@
     "watch-ts": "tsc -w",
     "watch-node": "nodemon dist/app.js",
     "watch": "concurrently -k -p \"[{name}]\" -n \"TypeScript, Node\" -c \"yello.bold, cyan.bold\" \"yarn run watch-ts\" \"yarn run watch-node\"",
-    "docs": "apidoc -i src/ -o public/apidoc/"
+    "docs": "apidoc -i src/ -o public/apidoc/",
+    "lint": "yarn eslint . --ext .js,.jsx,.ts,.tsx"
```
Comment (edudepetris) [inline package.json]: @jlgasparrini 
we need to reduce the scope of this, and as we are just an api on node we don't have jsx files. 
I would suggest this `yarn eslint --ext .js,.ts,.tsx src/** test/**`

File: .eslintrc.json (hunk 1 above)
Comment (jlgasparrini) [reply .eslintrc.json]: Yes I know it, this is just an autogenerated file. I'll change it

File: package.json [hunk 3]
```diff
@@ -27,17 +27,25 @@
...
   "devDependencies": {
     "@types/chai": "^4.1.5",
     "@types/mocha": "^5.2.5",
     "@types/mongoose": "^5.2.17",
     "@types/nock": "^9.3.0",
+    "@typescript-eslint/eslint-plugin": "^4.6.0",
+    "@typescript-eslint/parser": "^4.6.0",
     "chai": "^4.2.0",
+    "eslint": "^7.12.1",
+    "eslint-config-standard": "^16.0.0",
```
Comment (edudepetris) [inline package.json]: @jlgasparrini 
a fresh eslint installation and config (follow the eslint guide) these are packages are being included:
Do we have any reason for include others? 

```diff
diff --git a/package.json b/package.json
index b935a40..f00d9f8 100644
--- a/package.json
+++ b/package.json
@@ -34,7 +34,10 @@
     "@types/mocha": "^5.2.5",
     "@types/mongoose": "^5.2.17",
     "@types/nock": "^9.3.0",
+    "@typescript-eslint/eslint-plugin": "^4.6.0",
+    "@typescript-eslint/parser": "^4.6.0",
     "chai": "^4.2.0",
+    "eslint": "^7.12.1",
     "mocha": "^5.2.0",
     "nock": "^10.0.0",
     "ts-node": "^7.0.1",
```

**Comment by edudepetris** [comment]
after the configuration of eslint by cli, it reduces the scope of errors/warns to 24 💪 

I love this feature, it will bump up the code quality 💙 and open the doors to run it on github actions 🚀, prettier ... 

thanks for the contribution!

**Comment by jlgasparrini** [comment]
It's a pleasure! I already update those changes. I learned also that in the community exist other standards like Airbnb configuration. Taking a look [here](https://www.npmtrends.com/eslint-config-airbnb-vs-eslint-config-recommended-vs-eslint-config-standard) I saw that is very popular.

**Comment by edudepetris** [comment]
> It's a pleasure! I already update those changes. I learned also that in the community exist other standards like Airbnb configuration. Taking a look [here](https://www.npmtrends.com/eslint-config-airbnb-vs-eslint-config-recommended-vs-eslint-config-standard) I saw that is very popular.

well looks like the airbnb is the way to go, I'm happy with what ever you choose

**Comment by jlgasparrini** [comment]
I think that Airbnb’s style guide is generally considered the best option for React development. If you agree I preferred to work with the recommended plugin.

File: package.json (hunk 3 above)
Comment (edudepetris) [reply package.json]: shall we remove these as they are not included on a fresh eslint install?
"eslint-plugin-import": "^2.22.1",
"eslint-plugin-node": "^11.1.0",
"eslint-plugin-promise": "^4.2.1",

**Pull request merged**